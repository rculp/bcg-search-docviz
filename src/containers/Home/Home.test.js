import { shallow, mount } from 'enzyme';

import React from 'react';
import configureStore from 'redux-mock-store'
import { Input } from 'semantic-ui-react';

import ConnectedHomeContainer, { HomeContainer } from './Home';

describe('Home', () => {
  const mockStore = configureStore();
  const initialState = {
    search: {
      searchValue: '',
      loading: false,
      error: false,
      shouldRedirect: false
    }
  };
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('connects to redux store', () => {
    const component = shallow(
      <ConnectedHomeContainer store={store}/>
    );

    expect(component.props().searchValue).toEqual(initialState.search.searchValue);
    expect(component.props().loading).toEqual(initialState.search.loading);
    expect(component.props().error).toEqual(initialState.search.error);
    expect(component.props().shouldRedirect).toEqual(initialState.search.shouldRedirect);
  });

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <HomeContainer />
    );
    expect(component).toMatchSnapshot();
  });

  it('goes through lifecycle methods', () => {
    const component = mount(
      <HomeContainer />
    );
    //const componentDidUpdateSpy = jest.spyOn(component.prototype, 'componentDidUpdate');
    //expect(componentDidUpdateSpy).toHaveBeenCalled();
    expect(component.find(Input).simulate('change', {value: 'test'}));
  });
});
