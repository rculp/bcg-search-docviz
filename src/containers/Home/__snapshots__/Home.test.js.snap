// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders and matches our snapshot 1`] = `
<Page
  id="home"
>
  <main>
    <h2>
      Sinequa Search
    </h2>
    <Form
      as="form"
    >
      <FormField>
        <Input
          icon="search"
          iconPosition="left"
          id="query"
          label={
            <Button
              as="button"
              onClick={[Function]}
              role="button"
              type="submit"
            >
              Submit
            </Button>
          }
          labelPosition="right"
          onChange={[Function]}
          placeholder="Search..."
          size="massive"
          type="text"
        />
      </FormField>
    </Form>
  </main>
</Page>
`;
